cli/replicaset/cmd/upgrade.go

```diff
@@ -0,0 +1,272 @@
+package replicasetcmd
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/tarantool/tt/cli/connector"
+	"github.com/tarantool/tt/cli/replicaset"
+	"github.com/tarantool/tt/cli/running"
+)
+
+var (
+	// List of replicaset names specified by the user for the upgrade.
+	ChosenReplicasetAliases []string
+	// Timeout period (in seconds) for waiting on LSN synchronization.
+	LsnTimeout int
+)
+
+type instanceMeta struct {
+	run  running.InstanceCtx
+	conn connector.Connector
+}
+
+//go:embed lua/upgrade.lua
+var upgradeMasterLua string
+
+type syncInfo struct {
+	LSN uint64  `mapstructure:"lsn"`
+	IID uint32  `mapstructure:"iid"`
+	Err *string `mapstructure:"err"`
+}
+
+// Filters the given replicaset list by chosen aliases and returns the chosen
+// replicasets. If a non-existent alias is found, it returns an error.
+func filterReplicasetsByAliases(replicasets replicaset.Replicasets) ([]replicaset.Replicaset,
+	error) {
+	// If no aliases are provided, return all replicasets.
+	if len(ChosenReplicasetAliases) == 0 {
+		return replicasets.Replicasets, nil
+	}
+
+	// Create a map for fast lookup of replicasets by alias.
+	replicasetMap := make(map[string]replicaset.Replicaset)
+	for _, rs := range replicasets.Replicasets {
+		replicasetMap[rs.Alias] = rs
+	}
+
+	var chosenReplicasets []replicaset.Replicaset
+	for _, alias := range ChosenReplicasetAliases {
+		rs, exists := replicasetMap[alias]
+		if !exists {
+			return nil, fmt.Errorf("replicaset with alias %q doesn't exist", alias)
+		}
+		chosenReplicasets = append(chosenReplicasets, rs)
+	}
+
+	return chosenReplicasets, nil
+}
+
+func Upgrade(discoveryCtx DiscoveryCtx) error {
```

Comment:
Like this, as example.
```suggestion

func Upgrade(discoveryCtx DiscoveryCtx, opts UpgrateOpts) error {
```

Reply:
Done.